Changes

```diff
@@ -327,6 +327,9 @@ Working version
 
 ### Bug fixes:
 
+- #5673, #7636: unused type variable causes generalization error
+  (Jacques Garrigue, report by Christophe Raffalli)
```

Comment:
?? Added others involved.

Reply:
Well the code in this PR is almost verbatim from my comment: https://github.com/ocaml/ocaml/pull/1588#issuecomment-361557731. So I was expecting at least some credit in the Changes entry -- I will accept the review credit that you have since added.

Reply:
Right. I gradually converged to your code, removing all the extra bits I had in the beginning.
The only place you can recognize that I didn't copy it is that I was not using `exists2`, but even that has been fixed, so yes, this is your code.

Reply:
I just added Leo as an author in the Changes.